test/package.json

```diff
@@ -0,0 +1,4 @@
+{
+  "description": "Without this nodejs >=22.7.0 will see the module syntax in the tests, assume incorectly that we are using ESM, and fail once it tries run the imports",
```

Comment:
See: https://github.com/nodejs/node/blob/main/doc/changelogs/CHANGELOG_V22.md#module-syntax-detection-is-now-enabled-by-default

W/o this running the tests on node 22.7 fails with:
```
(node:93162) [MODULE_TYPELESS_PACKAGE_JSON] Warning: file:///<path>/odata-to-abstract-sql/test/chai-sql.js parsed as an ES module because module syntax was detected; to avoid the performance penalty of syntax detection, add "type": "module" to /<path>/odata-to-abstract-sql/test/package.json
(Use `node --trace-warnings ...` to show where the warning was created)

 Exception during run: file:///<path>/odata-to-abstract-sql/test/chai-sql.js:5
import { createTranslator } from '@balena/lf-to-abstract-sql';
         ^^^^^^^^^^^^^^^^
SyntaxError: Named export 'createTranslator' not found. The requested module '@balena/lf-to-abstract-sql' is a CommonJS module, which may not support all module.exports as named exports.
CommonJS modules can always be imported via the default export, for example using:

import pkg from '@balena/lf-to-abstract-sql';
const { createTranslator } = pkg;

    at ModuleJob._instantiate (node:internal/modules/esm/module_job:171:21)
    at async ModuleJob.run (node:internal/modules/esm/module_job:254:5)
    at async onImport.tracePromise.__proto__ (node:internal/modules/esm/loader:482:26)
    at async formattedImport (/<path>/odata-to-abstract-sql/node_modules/mocha/lib/nodejs/esm-utils.js:9:14)
    at async Object.exports.requireOrImport (/<path>/odata-to-abstract-sql/node_modules/mocha/lib/nodejs/esm-utils.js:42:28)
    at async Object.exports.loadFilesAsync (/<path>/odata-to-abstract-sql/node_modules/mocha/lib/nodejs/esm-utils.js:100:20)
    at async singleRun (/<path>/odata-to-abstract-sql/node_modules/mocha/lib/cli/run-helpers.js:162:3)
    at async Object.exports.handler (/<path>/odata-to-abstract-sql/node_modules/mocha/lib/cli/run.js:375:5)
```

Reply:
Yeah the real issue is that a `type` should be present in the top-level package.json which for this repo is indeed commonjs, I've fixed that and most of the lint warnings in https://github.com/balena-io-modules/odata-to-abstract-sql/pull/155 so rebasing on top of that should remove a lot of the noise from this PR (mainly due to the linting warnings)

Reply:
Oh, I would swear that I also tried adding the type to the top level package.json... Well we indeed don't need this now :+1: